R/writeMSData.R

```diff
@@ -44,58 +44,68 @@
     x_split <- splitByFile(x, f = factor(fileNames(x)))
     ## Using mapply below - in principle we could then even switch to
     ## bpmapply to perform parallel saving of files.
-    res <- mapply(x_split, files, FUN = function(z, f, outformat,
-                                                 software_processing,
-                                                 verbose){
-        if (verbose)
-            message("Saving file ", basename(f), "...", appendLF = FALSE)
-        if (file.exists(f))
-            stop("File ", f, " does already exist! Please use a different ",
-                 "file name.")
-        ## o get all peak data. Calling spectrapply will apply also all lazy
-        ##   processing steps. We're not calling the `as` method to avoid R
-        ##   having to look through all namespaces to find the function.
-        pks <- spectrapply(z, FUN = function(sp) cbind(sp@mz, sp@intensity),
-                           BPPARAM = SerialParam())
-        if (is(x, "OnDiskMSnExp")) {
-            hdr <- fData(z)
+    res <- mapply(FUN = .writeSingleMSData, x_split, files,
```

Comment:
The assignment to `res` is not needed here. Just `mapply(...) }`, so the return value of `mapply` will be the return value of `.writeMSData`.

Reply:
I don't want to return anything - so I'll use `invisible` instead.

Reply:
Ok, I thought `.writeSingleMSData` (or `mzR::writeMSData`) would return a value for success/error. But if not `invisible` is fine.